src/App.js

```diff
@@ -19,48 +19,50 @@ export default function App() {
     saveToLocal('bookmarkedIds', bookmarkedIds)
   }, [bookmarkedIds])
 
+  const { push } = useHistory()
+
   return (
     <AppGrid>
-      {activePage === 'LocationPage' && (
-        <LocationPage
-          onDetail={showDetail}
-          bookmarkedIds={bookmarkedIds}
-          locations={locations}
-          toFavorite={toFavorite}
-          handleBookmark={handleBookmark}
-        />
-      )}
-
-      {activePage === 'DetailsPage' && (
-        <DetailsPage
-          onNavigate={handleClickBack}
-          toFavorite={toFavorite}
-          details={details}
-        />
-      )}
-
-      {activePage === 'FavoritePage' && (
-        <FavoritePage
-          onNavigate={handleClickBack}
-          locations={locations}
-          bookmarkedIds={bookmarkedIds}
-          handleBookmark={handleBookmark}
-        />
-      )}
+      <Switch>
+        <Route exact path="/">
+          <LocationPage
+            onDetail={showDetail}
+            bookmarkedIds={bookmarkedIds}
+            locations={locations}
+            toFavorite={toFavorite}
+            handleBookmark={handleBookmark}
+          />
+        </Route>
+        <Route path="/DetailsPage">
+          <DetailsPage
+            onNavigate={handleClickBack}
+            toFavorite={toFavorite}
+            details={details}
+          />
+        </Route>
+        <Route path="/FavoritePage">
+          <FavoritePage
+            onNavigate={handleClickBack}
+            locations={locations}
+            bookmarkedIds={bookmarkedIds}
+            handleBookmark={handleBookmark}
+          />
+        </Route>
+      </Switch>
     </AppGrid>
   )
 
   function showDetail(id) {
-    setActivePage('DetailsPage')
     setDetailsId(id)
+    push('/DetailsPage')
   }
 
   function handleClickBack() {
-    setActivePage('LocationPage')
+    push('/')
   }
 
   function toFavorite() {
-    setActivePage('FavoritePage')
+    push('/FavoritePage')
```

Comment:
This is just a sugegstion :)
You could implement a Link from react-router-dom in the component where these functions get used to link to the page. like:
`<Link to='/' />`
You would save yourself some code and also some props :)

Reply:
hey thanks Mel , yes i wanna do this in the next step, when i implement a navigation to my app